pkg/controller/s3/bucket.go

```diff
@@ -98,43 +98,45 @@ func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 	}
 
 	if _, err := e.s3client.HeadBucketRequest(&awss3.HeadBucketInput{Bucket: aws.String(meta.GetExternalName(cr))}).Send(ctx); err != nil {
-		return managed.ExternalObservation{}, awsclient.Wrap(resource.Ignore(s3.IsNotFound, err), errHead)
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(s3.IsNotFound, err), errHead)
```

Comment:
Should this definitely change? iirc `awsclient.Wrap` extends `errors.Wrap` to trim UIDs from AWS errors so that we don't consider two instances of the same error to be two different errors.